core/src/main/java/com/homeaway/streamplatform/streamregistry/resource/ClusterResource.java

```diff
@@ -0,0 +1,177 @@
+        @ApiResponse(code = 500, message = "Error Occurred while getting data"),
+        @ApiResponse(code = 404, message = "Cluster not found") })
+    @Produces(MediaType.APPLICATION_JSON)
+    @Timed
+    public Response getAllClusters(@ApiParam(value = "vpc") @QueryParam("vpc") String vpc,
+        @ApiParam(value = "env") @QueryParam("env") String env,
+        @ApiParam(value = "hint") @QueryParam("hint") String hint,
+        @ApiParam(value = "type") @QueryParam("type") String type) {
+        try {
+            Map<ClusterKey, ClusterValue> clusterMap = clusterDao.getAllClusters();
+
+            if(vpc != null && !vpc.isEmpty()) {
+                clusterMap = clusterMap.entrySet().stream()
+                    .filter(e -> e.getKey().getVpc().equalsIgnoreCase(vpc))
+                    .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+            }
+
+            if(env != null && !env.isEmpty()) {
+                clusterMap = clusterMap.entrySet().stream()
+                    .filter(e -> e.getKey().getEnv().equalsIgnoreCase(env))
+                    .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+            }
+
+            if(hint != null && !hint.isEmpty()) {
+                clusterMap = clusterMap.entrySet().stream()
+                    .filter(e -> e.getKey().getHint().equalsIgnoreCase(hint))
+                    .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+            }
+
+            if(type != null && !type.isEmpty()) {
+                clusterMap = clusterMap.entrySet().stream()
+                    .filter(e -> e.getKey().getType().equalsIgnoreCase(type))
+                    .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+            }
+
+            return Response.status(200).entity(clusterMap).build();
+        } catch (Exception e) {
+            log.error("Error getting cluster details.", e);
+            return Response.status(Response.Status.INTERNAL_SERVER_ERROR)
+                .entity(new ErrorMessage(Response.Status.INTERNAL_SERVER_ERROR.getStatusCode(),
+                    "Error getting cluster details.",
+                    e.getMessage()))
+                .build();
+        }
+    }
+
+    @GET
+    @Path("/{clusterName}")
+    @ApiOperation(
+        value = "Get a Cluster",
+        notes = "Returns a single cluster",
+        tags = "clusters",
+        response = ClusterValue.class)
+    @ApiResponses(value = { @ApiResponse(code = 200, message = "Returns Cluster information", response = ClusterValue.class),
+        @ApiResponse(code = 500, message = "Error Occurred while getting data"),
+        @ApiResponse(code = 404, message = "Cluster not found") })
+    @Produces(MediaType.APPLICATION_JSON)
+    @Timed
+    public Response getCluster(@ApiParam(value = "clusterName", required = true) @PathParam("clusterName") String clusterName) {
```

Comment:
Get by clustername - ClusterName is currently not part of the key

Reply:
however, cant we just add `clusterName` as a query param in `getAllClusters` , and do the same filtering that has been done in this method.